core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -445,6 +479,25 @@ private SaltConfig createSaltConfig(Stack stack, Cluster cluster, List<GrainProp
         return new SaltConfig(servicePillar, grainsProperties);
     }
 
+    private SaltConfig createSaltConfigWithGatewayPillarOnly(Stack stack, Cluster cluster, List<GrainProperties> grainsProperties)
+            throws IOException, CloudbreakOrchestratorException {
+
+        GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
+        String virtualGroupsEnvironmentCrn = environmentConfigProvider.getParentEnvironmentCrn(stack.getEnvironmentCrn());
+        ClusterPreCreationApi connector = clusterApiConnectors.getConnector(cluster);
+        Map<String, List<String>> serviceLocations = getServiceLocations(cluster);
+        Optional<LdapView> ldapView = ldapConfigService.get(stack.getEnvironmentCrn(), stack.getName());
+        VirtualGroupRequest virtualGroupRequest = getVirtualGroupRequest(virtualGroupsEnvironmentCrn, ldapView);
+        KerberosConfig kerberosConfig = kerberosConfigService.get(stack.getEnvironmentCrn(), stack.getName()).orElse(null);
+        ClouderaManagerRepo clouderaManagerRepo = clusterComponentConfigProvider.getClouderaManagerRepoDetails(cluster.getId());
+
+        Map<String, SaltPillarProperties> servicePillar =
+                new HashMap<>(createGatewayPillar(primaryGatewayConfig, cluster, stack, virtualGroupRequest, connector, kerberosConfig, serviceLocations,
+                        clouderaManagerRepo));
+
+        return new SaltConfig(servicePillar, grainsProperties);
+    }
```

Comment:
can we have some debug logs for the seemingly major "events" of this creation process?

Reply:
ok